launcher/sdw_updater_gui/UpdaterApp.py

```diff
@@ -158,6 +186,48 @@ def exit_launcher(self):
         sys.exit()
 
 
+def _get_update_vm() -> str:
+    """
+    Helper function to query qubes-prefs for the designated UpdateVM.
+    """
+    updatevm = None
+
+    try:
+        updatevm_bytes = subprocess.check_output(["qubes-prefs", "updatevm"])
+        if updatevm_bytes:
+            updatevm = updatevm_bytes.decode("utf-8").strip()
+        else:
+            logger.error("qubes-prefs returned null value for UpdateVM")
+    except subprocess.CalledProcessError as e:
+        logger.error("Failed to complete UpdateVM query")
+        logger.error(str(e))
+
+    return updatevm
+
+
+def _is_updatevm_connected(updatevm: str) -> bool:
+    """
+    Helper function to assess network connectivity before launching updater.
+
+    Assess network connectivity by attempting to reach one of sites used for
+    package updates.
+    """
+    curl_commands = [
+        b"curl -fsI https://mirrors.fedoraproject.org --connect-time 2 --max-time 5",
+        b"curl -fsI https://apt.freedom.press --connect-time 2 --max-time 6",
+        b"curl -fsI https://ftp.qubes-os.org --connect-time 3 --max-time 8",
+    ]
```

Comment:
This part is open to discussion. I think trying all 3 domains is overkill, 2 is probably sufficient to say we have connection problems (noting that more than one request will only be made if the first one fails). Also welcome feedback on timeout lengths.

Reply:
@rocodes How about:

```
qvm-run -p sys-net 'nmcli --terse -f state general'
```

That will output a string "connected" or "disconnected", depending on state. I've only tested it with wifi, but it should work equally well for ethernet, too.

Reply:
much improved, thank you :)  changed in  9c5b5e5

Reply:
I did some digging and I'm not sure that `nmcli -f state general` differentiates between connected locally (but no internet access) and connected (with internet access), whereas `nmcli networking connectivity check` will check immediately, and report back `full`, `limited`, `none`, [etc](https://developer-old.gnome.org/NetworkManager/stable/nmcli.html) (so if a user is connected to a network with a captive portal, the status will be `portal`).